.github/workflows/ci.yml

```diff
@@ -151,13 +151,20 @@ jobs:
         run: cd build && make flake8
       - id: files
         uses: Ana06/get-changed-files@v2.1.0
+        continue-on-error: true
       - name: clang-tidy
         run: |
+          if [[ -n "${{ steps.files.outputs.all }}" ]]; then
+            FILES="${{ steps.files.outputs.all }}"
+          else
+            git fetch origin main && git checkout -b main origin/main || true
+            FILES=$(git diff --name-only main ${{ github.sha }} --)
```

Comment:
so on first PR action run this will run clang-tidy for all changed files (relative to main) and then next runs will run clang-tidy for all files relative to the previous commit?

Reply:
Yes, at least on push triggers.

Reply:
would it make sense to always check relative to main? I'm not sure what the average time clang-tidy takes

Reply:
Well it's relative to the number of cpp files changed, so it can be long if we do it for every single commit. I would stick with checking only the newly changed files per pushes, as we will run on every file eventually on PRs.

Reply:
makes sense